crates/pink/pink-extension/src/chain_extension.rs

```diff
@@ -104,105 +107,520 @@ pub type BatchHttpResult = Result<Vec<Result<HttpResponse, HttpRequestError>>, H
+    /// let response = pink::ext().http_request(request);
+    /// ```
+    ///
+    /// There are also some shortcut macros for this function:
+    /// - [`http_get!`]
+    /// - [`http_post!`]
+    /// - [`http_put!`]
+    ///
+    /// # Availability
+    /// any contract | query only
     #[ink(extension = 1, handle_status = false)]
     fn http_request(request: HttpRequest) -> HttpResponse;
 
+    /// Sign a message with a given key.
+    ///
+    /// # Arguments
+    ///
+    /// * `sigtype`: The signature type to use for signing the message.
+    /// * `key`: The private key used for signing the message.
+    /// * `message`: The message to be signed.
+    ///
+    /// # Returns
+    ///
+    /// * `Vec<u8>` - The signed message as a byte vector.
+    ///
+    /// # Example
+    ///
+    /// ```ignore
+    /// let derived_key = pink::ext().derive_sr25519_key(b"some salt".into());
+    /// let message = b"Hello, world!";
+    /// let signature = pink::ext().sign(SigType::Sr25519, &key, message);
+    /// let pubkey = pink::ext().get_public_key(SigType::Sr25519, &derived_key);
+    /// let is_valid = pink::ext().verify(SigType::Sr25519, &pubkey, message, &signature);
+    /// ```
+    ///
+    /// # Availability
+    /// For SigType::Sr25519:
+    ///     any contract | query only
+    ///
+    /// For Others:
+    ///     any contract | query | transaction
     #[ink(extension = 2, handle_status = false)]
     fn sign(sigtype: SigType, key: &[u8], message: &[u8]) -> Vec<u8>;
 
+    /// Verify a signature.
+    ///
+    /// This method verifies a digital signature given the signature type, public key, message, and signature.
+    ///
+    /// # Arguments
+    ///
+    /// * `sigtype`: The type of signature to verify.
+    /// * `pubkey`: The public key associated with the private key that signed the message.
+    /// * `message`: The original message that was signed.
+    /// * `signature`: The digital signature to verify.
+    ///
+    /// # Returns
+    ///
+    /// * `bool` - `true` if the signature is valid, `false` otherwise.
+    ///
+    /// # Example
+    ///
+    /// ```ignore
+    /// let derived_key = pink::ext().derive_sr25519_key(b"some salt".into());
+    /// let message = b"Hello, world!";
+    /// let signature = pink::ext().sign(SigType::Sr25519, &key, message);
+    /// let pubkey = pink::ext().get_public_key(SigType::Sr25519, &derived_key);
+    /// let is_valid = pink::ext().verify(SigType::Sr25519, &pubkey, message, &signature);
+    /// ```
+    ///
+    /// # Availability
+    /// any contract | query | transaction
     #[ink(extension = 3, handle_status = false)]
     fn verify(sigtype: SigType, pubkey: &[u8], message: &[u8], signature: &[u8]) -> bool;
 
+    /// Derive a sr25519 key.
+    ///
+    /// This method derives a sr25519 key using the provided salt and the contract private key.
+    /// The derived key is deterministic so it could be used in transactions to sign messages.
```

Comment:
Let's also mention it can be used as a cryptographically secure entropy source

Reply:
Added